Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -307,7 +310,8 @@ void GroupDetectors2::exec() {
            : 1.);
 
   // Build a new map
-  const size_t outIndex = formGroups(inputWS, outputWS, prog4Copy);
+  const size_t outIndex =
+      formGroups(inputWS, outputWS, prog4Copy, keepAll, unGroupedSet);
 
   // If we're keeping ungrouped spectra
   if (keepAll) {
```

Comment:
`moveOthers` under the hood sets the output spectra to the input spectra:
```
outputWS.getSpectrum(outIndex) = inputWS.getSpectrum(sourceIndex);
```

This I guess would overwrite the grouping, but should not change the spectrum numbers you end up with.

I have move the setting of `IndexInfo` to after `moveOthers`, as I think that should make more sense.